### ISSUE TITLE
Round and square brackets highlighting stop working when starting edit the file in C#

### ISSUE THREAD

**Issue by Streuselkuchen** [izhangzhihao/intellij-rainbow-brackets#2569]
## Please check

- [ ] I already support this project.
- [ ] If you are submitting a feature request, please do consider buy this plugin [here](https://www.jetbrains.com/shop/buy?item=P:N:PRAINBOWBRACKET:M) or donating us on [Open Collective](https://opencollective.com/intellij-rainbow-brackets) Or by AliPay/WeChatPay.
- [x] This issue/feature request is not reported before.

## Your programming languages
C#

## Expected Behavior
When I open new round Brackets in existing round Brackets they should have a different color.

## Current Behavior
All Round Brackets have the same color, and it doesn't matter which level they have.

## Possible Solution
/

## Code snippet for reproduce (for bugs)
![image](https://user-images.githubusercontent.com/86515910/206403563-f7344c66-89f7-4706-b3cc-5a97af7ad65f.png)

## Your Environment

* Plugin version: 
   2022.3.5
* IDE & Operating System version, comment your env as below(go to "About IntelliJ IDEA" -> click the "copy" icon):
```
JetBrains Rider 2022.3
Build #RD-223.7571.232, built on December 5, 2022
Licensed to M. D.
Subscription is active until January 6, 2023.
Evaluation purpose only.
Runtime version: 17.0.5+1-b653.14 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
.NET 7.0.0 (Server GC)
GC: G1 Young Generation, G1 Old Generation
Memory: 1500M
Cores: 16
Registry:
    debugger.new.tool.window.layout=false
    platform.projectModel.workspace.model.file.index=false
    ide.new.project.model.index.case.sensitivity=true
    database.show.search.tab=false
    indexing.enable.entity.provider.based.indexing=false
    vcs.empty.toolwindow.show=false
    rdclient.asyncActions=false
    ide.workspace.model.rbs.as.tree=false

Non-Bundled Plugins:
    lermitage.intellij.extratci (1.24.0)
    com.intellij.resharper.HeapAllocationsViewer (2022.3.0-eap04)
    com.kagof.pokeprogress (2.1.1)
    cognitivecomplexity-rider (2022.3.0-eap04)
    Key Promoter X (2022.3)
    com.intellij.resharper.StructuredLogging (2022.3.0.276-eap07)
    com.mallowigi.colorHighlighter (14.0.0)
    aws.toolkit (1.55-223)
    izhangzhihao.rainbow.brackets (2022.3.5)

```

**Comment by izhangzhihao** [comment]
Hi, it works for me as the screenshots:
<img width="443" alt="image" src="https://user-images.githubusercontent.com/12044174/206620092-6b133483-0dec-4427-8270-47115e0ac013.png">


```c#
namespace DefaultNamespace
{
    public class A
    {
        private void test()
        {
            bool a = true;
            bool b = true;
            bool c = true;
            bool d = true;
            
			if (a == b && (c || (d && e))) {
			
			}
        }
    }
}
```

**Comment by hwooo** [comment]
In my Rider 2022.3, rainbow variables option is not working at all.

![screenshot](https://user-images.githubusercontent.com/46282492/206824973-9a445f3d-3de9-4492-9e01-efa9f692f0f6.png)

**Comment by izhangzhihao** [comment]
Hi @hwooo I created #2570 for your request~

**Comment by rherzfeld** [comment]
It's not working for me either. Possibly because I don't have a license (I get a banner that the language is not supported).
However, it was working before and I would expect rainbow brackets to be a language independent core feature, given it's the name of the plugin.

**Comment by izhangzhihao** [comment]
hi @rherzfeld , I can confirm that these features are not available because they have not been activated.

The reason for this is that since the introduction of break changes by Rider in 2022.3, C#-related features have required a significant amount of rewriting in order to fix existing bugs and improve their functionality. As a result, these features have been marked as paid features starting in 2022.3

However, you are welcome to try these features for free for 30 days before deciding whether or not to purchase them.

**Comment by izhangzhihao** [comment]
a workaround for this issue, use ctrl-a and ctrl-x to cut all content from the current file and restore all content to the file again using ctrl-v, and all seems to look fine.

to repeat this issue, just close and re-open ide and start editing the file.

I think this is caused by the non-deterministic of the parser, I will report an issue on YouTrack(https://youtrack.jetbrains.com/) later.

**Comment by AlanDavison** [comment]
> hi @rherzfeld , I can confirm that these features are not available because they have not been activated.
> 
> The reason for this is that since the introduction of break changes by Rider in 2022.3, C#-related features have required a significant amount of rewriting in order to fix existing bugs and improve their functionality. As a result, these features have been marked as paid features starting in 2022.3
> 
> However, you are welcome to try these features for free for 30 days before deciding whether or not to purchase them.

I will point out that you state C# is supported on the free lite version of the plugin on the plugin's page. You might want to change that before you get some very irate people complaining you're misleading them.

**Comment by izhangzhihao** [comment]
I created a report in YouTrack [Highlightings for C# code disappear when editing code](https://youtrack.jetbrains.com/issue/RIDER-87202/Highlightings-for-C-code-disappear-when-editing-code)

**Comment by eldaville** [comment]
I have the same problem

This round brackets should be colored, using Rider 2022.3.1 

![image](https://user-images.githubusercontent.com/34583909/211968014-59ccc3a2-d8da-4287-9c45-f9e8b4cc8fc4.png)

**Comment by eldaville** [comment]
When you just open Rider, the plugin colors correctly the ( ) but once you edit anything it changes their colors to the be the same color... it only displays the expected colors when you close and open again Rider... it doesn't gets fixed if you just close / open the file.

![image](https://user-images.githubusercontent.com/34583909/213543559-2ca158eb-7e3b-4f12-896f-8c4c80f6d69b.png)

**Comment by izhangzhihao** [comment]
[The JB official said that "this issue has been added to our list of 2023.3 candidates(of the Rider IDE release). It's not guaranteed to be fixed, but we are looking into it"](https://youtrack.jetbrains.com/issue/RIDER-87202/Highlightings-for-C-code-disappear-when-editing-code#focus=Comments-27-7369993.0-0)

So, let's keep patience.

**Comment by izhangzhihao** [comment]
Hi, this bug seems fixed in the latest EAP: **JetBrains Rider 2023.2 EAP 5**.

I tried to edit these files and the highlights keep stable.

<img width="584" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/0d783b38-746f-44d4-ba8d-9887d41a06cc">

<img width="518" alt="image" src="https://github.com/izhangzhihao/intellij-rainbow-brackets/assets/12044174/21921f28-e48c-4cd0-afd4-8bbb13b9d3ba">
